storage/innobase/include/buf0buf.h

```diff
@@ -657,6 +657,22 @@ class buf_page_t
     access_time= 0;
   }
 
+private:
+  void set_memory_state(uint32_t s)
+  {
+#ifdef MADV_FREE
+    if (s == buf_page_t::NOT_USED)
+      madvise(frame, srv_page_size, MADV_FREE);
+#endif
+#ifdef _WIN32
```

Comment:
Shouldn’t it be `#elif defined _WIN32`? We can’t use both interfaces at the same time.